content/en/blog/_posts/2023-11-20-volume-attributes-class/index.md

```diff
@@ -0,0 +1,180 @@
+layout: blog
+title: "Kubernetes 1.29: VolumeAttributesClass for Volume Modification"
+date: 2023-12-01
+slug: kubernetes-1-29-volume-attributes-class
+---
+
+**Authors**: Sunny Song(Google)
+
+The v1.29 release of Kubernetes introduced an alpha feature to support modify volume
+by changing VolumeAttributesClass that was assigned to a PersistentVolumeClaim (PVC).
+With the feature enabled, Kubernetes can handle updates of volume attributes other than capacity.
+Allowing volume attributes to be changed without managing it through different
+provider's APIs directly simplifies the current flow.
+
+You can read VolumeAttributesClass in the Kubernetes documentation for more details
+about how to use that, or you can read on to learn about why the Kubernetes project is supporting this feature.
+
+
+## VolumeAttributesClass
+
+The new resource.k8s.io/v1alpha1 API group provides two new types:
+
+**VolumeAttributesClass**
+
+Represents a specification of mutable volume attributes defined by the CSI driver.
+The class can be specified during dynamic provisioning of PersistentVolumeClaims,
+and changed in the PersistentVolumeClaim spec after provisioning.
+
+**ModifyVolumeStatus**
+
+Represents the status object of ControllerModifyVolume operation.
+
+With this alpha feature enabled, the spec of PersistentVolumeClaim defines VolumeAttributesClassName
+that are used in the PVC. At volume provisioning, CreateVolume operation will apply the parameters in the
+VolumeAttributesClass along with the parameters in the StorageClass.
+
+When there is a change of volumeAttributesClassName in the PVC spec,
+the external-resizer sidecar will get an informer event. Based on the current state of the configuration,
+the resizer will trigger a CSI ControllerModifyVolume. The diagram below shows the flow:
+
+![VolumeAttributesClass ModifyVolume Flow](VolumeAttributesClass-ModifyVolume-Flow-v3.png)
```

Comment:
Please add an alt attribute that explains the diagram to someone who who has trouble seeing clearly. Ideally: use a [`figure`](https://gohugo.io/content-management/shortcodes/#figure) shortcode.